### ISSUE TITLE
 ので is split into の and で, which it should not in my opinion

### ISSUE THREAD

**Issue by TwoDeen** [mkartawijaya/dango#2]
ので is split into の and で, which it should not in my opinion and determine it as 接続詞 CONJUNCTION

For this sentence "明日は早く起きなければならないので、今日は早めに寝ます。"  I get the following output
NOUN 明日 あす 明日 あす
PARTICLE は は は は
ADJECTIVE 早く はやく 早い はやい
VERB 起きなけれ おきなけれ 起きる おきる
PARTICLE ば ば ば ば
VERB ならない ならない なる なる
**PARTICLE の の の の
AUXILIARY_VERB で で だ だ**
SYMBOL 、 、 、 、
NOUN 今日 きょう 今日 きょう
PARTICLE は は は は
ADJECTIVAL_NOUN 早め はやめ 早め はやめ
AUXILIARY_VERB に に だ だ
VERB 寝ます ねます 寝る ねる
SYMBOL 。 。 。 。

**Comment by mkartawijaya** [comment]
I would agree with you that the ので should not be split here.

However, dango itself does not do any splitting. It works by taking the tokenized output of the SudachiPy morphological analyzer and aggregating it into "words" based on a set of rules, mostly by looking at the part-of-speech information.

One has to be careful to not make the rules too generic since this could cause tokens to be aggregated that should not be. Take for example the phrase 赤いのでお願いします。Here we would want the ので to be split but the analyzer's output regarding the の and で tokens is the same as in your case.

```
明日	名詞,普通名詞,副詞可能,*,*,*	明日
は	助詞,係助詞,*,*,*,*	は
早く	形容詞,一般,*,*,形容詞,連用形-一般	早い
起き	動詞,一般,*,*,上一段-カ行,未然形-一般	起きる
なけれ	助動詞,*,*,*,助動詞-ナイ,仮定形-一般	ない
ば	助詞,接続助詞,*,*,*,*	ば
なら	動詞,非自立可能,*,*,五段-ラ行,未然形-一般	成る
ない	助動詞,*,*,*,助動詞-ナイ,連体形-一般	ない
の	助詞,準体助詞,*,*,*,*	の
で	助動詞,*,*,*,助動詞-ダ,連用形-一般	だ
、	補助記号,読点,*,*,*,*	、
今日	名詞,普通名詞,副詞可能,*,*,*	今日
は	助詞,係助詞,*,*,*,*	は
早め	形状詞,一般,*,*,*,*	早め
に	助動詞,*,*,*,助動詞-ダ,連用形-ニ	だ
寝	動詞,一般,*,*,下一段-ナ行,連用形-一般	寝る
ます	助動詞,*,*,*,助動詞-マス,終止形-一般	ます
。	補助記号,句点,*,*,*,*	。
```

```
赤い	形容詞,一般,*,*,形容詞,連体形-一般	赤い
の	助詞,準体助詞,*,*,*,*	の
で	助動詞,*,*,*,助動詞-ダ,連用形-一般	だ
お	接頭辞,*,*,*,*,*	御
願い	動詞,非自立可能,*,*,五段-ワア行,連用形-一般	願う
し	動詞,非自立可能,*,*,サ行変格,連用形-一般	為る
ます	助動詞,*,*,*,助動詞-マス,終止形-一般	ます
。	補助記号,句点,*,*,*,*	。
```

Can you provide any more cases where you (don't) want ので to be split? If so I could take a look at the analyzer's output to see if it is possible to differentiate the cases.

**Comment by TwoDeen** [comment]
I agree with your example. Mecab also suffers from the same error for your sentence where it’s not splitting “no” and “de”.
